sql/core/src/main/scala/org/apache/spark/sql/catalog/interface.scala

```diff
@@ -64,12 +64,26 @@ class Database(
 @Stable
 class Table(
     val name: String,
-    @Nullable val database: String,
+    @Nullable val qualifier: Array[String],
     @Nullable val description: String,
     val tableType: String,
     val isTemporary: Boolean)
   extends DefinedByConstructorParams {
 
+  def database: String = parseQualifier
+
+  def parseQualifier: String = {
+    if (qualifier == null) {
+      null
+    } else if (qualifier.length == 2) {
+      qualifier(1)
+    } else if (qualifier.length == 1) {
```

Comment:
when will `qualifiers` be length 1?

Reply:
for example if a table is `catalog1.db1.table1`,

In this case `catalog_name=catalog1, qualifier=[db1], table_name=table1`?